pkg/gadget-collection/gadgets/advise/seccomp/gadget.go

```diff
@@ -345,7 +345,13 @@ func (t *Trace) containerTerminated(trace *gadgetv1alpha1.Trace, event container
 
 	namespacedName := fmt.Sprintf("%s/%s", event.Container.Namespace, event.Container.Podname)
 
-	r, err := generateSeccompPolicy(t.client, trace, b, event.Container.Podname, event.Container.Name, namespacedName, event.Container.OwnerReference)
+	ownerRefernce, err := event.Container.GetOwnerReference()
+	if err != nil {
+		return
+	}
```

Comment:
The runtime-enrichment approach was also my first idea when working on this. However, after discussing with @alban, he made me realize that it could not work where we generate the seccomp-profile when the pod is deleted, because by the time we call `event.Container.GetOwnerReference()`, the pod could have been already deleted so we will not be able to enrich it.

Reply:
I see. I think this can be modified to enrich the container when it's created then, right?

Reply:
Yeah, but then what would be the difference to the current approach? IIUC, this PR is proposing to not enrich the container at creation but on-demand (when the seccomp gadget ask for it). Correct?

Reply:
We'll only enrich the containers that the seccomp advisor is interested in and not all of them. Also, they won't be enriched when Inspektor Gadget is started but when an instance of the seccomp advisor is created.

Reply:
Aahh ok, got it now. Yeah, it should work!

Reply:
I pushed a version with that approach. Please let me know if that looks better.